### PR TITLE
Export post permalink component

### PR DESCRIPTION
**Pull request by igmoweb**
`PostPermalink` in `@wordpress/editor` is not exported. We are developing a custom post title block and it would be handy to have this component exported instead of replicating it in our own codebase.

Related: #3517

### CODE REVIEW

File: packages/editor/src/components/index.js [hunk 1]
```diff
@@ -33,6 +33,7 @@ export { default as PostLastRevisionCheck } from './post-last-revision/check';
 export { default as PostLockedModal } from './post-locked-modal';
 export { default as PostPendingStatus } from './post-pending-status';
 export { default as PostPendingStatusCheck } from './post-pending-status/check';
+export { default as PostPermalink } from './post-permalink';
```
Comment (jorgefilipecosta) [inline packages/editor/src/components/index.js]: Thank you for your contribution @igmoweb 👍 It seems fine to expose this component. Would it be possible to add a changelog entry specifying that this component is now exposed (new features section)?

Comment (igmoweb) [reply packages/editor/src/components/index.js]: @jorgefilipecosta Added. I hope it's fine this way.

**Review by tfrommen — APPROVED** [review]
🚀

**Comment by gziolo** [comment]
@youknowriad - can you perform sanity check before we merge?

**Comment by youknowriad** [comment]
Now, that I think about it. We discussed removing that component entirely at some point since it's possible to edit the permalink in the sidebar.

Merging this PR would make that removal harder. I mean we'd still have a `PostPermalink` component used in the sidebar but design wise, it would be different. Would a different design count as a breaking change or not (for a component)?

**Comment by tfrommen** [comment]
I guess you are talking about the [`PostLink` component](https://github.com/WordPress/gutenberg/blob/master/packages/edit-post/src/components/sidebar/post-link/index.js) in the `edit-post` package, right?

Hm, if you are planning to remove the `PostPermalink` component eventually, then exposing it right now doesn't make much sense, yes.

That said—and answering you question about breaking changes—if you start with two things that allow to do (almost) the same thing, and then remove one of these things, to me, this is a breaking change anyway. At least from a user's perspective. Moving something from a dynamic inline block element to the Document sidebar is not just a re-design, in my opinion. And this is even more different on mobile...

**Comment by jorgefilipecosta** [comment]
Hi @youknowriad, @mapk, it seems until now we did not remove the sidebar permalink editor. In that case, should we merge this PR or the plan is to still remove the component from the sidebar?

**Comment by youknowriad** [comment]
I still think we should probably remove that component (the duplicate permalink UI above the title) but I'd like design confirmation @jasmussen @mapk @mtias

**Comment by jasmussen** [comment]
> I still think we should probably remove that component (the duplicate permalink UI above the title) but I'd like design confirmation

The permalink UI being attached to the Title has not been very discoverable in practice. So I would think that this component should either be rethought entirely, or we should remove it entirely in favor of just the permalink editing happening in the sidebar, and in the pre-publish flow (where the permalink isn't yet, but it probably should be). 

Additionally there's the caveat of the Title becoming an actual _block_ that you can remove entirely from a page, necessitating the ability to name the page and the permalink elsewhere (i.e. sidebar, on page creation, or [in the editor header bar if we can fit it](https://github.com/WordPress/gutenberg/issues/11553#issuecomment-457098878)).

**Comment by mapk** [comment]
Yep, removing it would be the future-forward path. The Title will be a block and so we'll need to add this permalink and title elsewhere in the UI.

**Comment by youknowriad** [comment]
Closing as this component has been removed.

**Pull request closed without merging**